### PR TITLE
feat: revamp create giftcard

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**

- Adds Create Giftcard Modal
- Minor update to FileUpload component to align it better with the design

image

image

image

**Depends on these PR's being merged beforehand**
- #260 
- #262

### CODE REVIEW

File: src/domain/gift-cards/index.tsx [hunk 1]
```diff
@@ -1,85 +1,77 @@
...
-}
+  const deleteGC = () => {
+    deleteGiftCard.mutate(undefined, {
+      onSuccess: () => navigate("/a/gift-cards"),
+    })
+  }
 
-const GiftCard = () => {
   return (
     <Router>
```
Comment (pKorsholm) [inline src/domain/gift-cards/index.tsx:60]: Router should be full height to better align with the design
```suggestion
    <Router className="h-full">
```

File: src/domain/gift-cards/new.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  }
+
+  const handleFileUpload = async (files) => {
+    setUploading(true)
+    //TODO upload files
+    await new Promise((r) => setTimeout(r, 2000))
+    setUploading(false)
+  }
+
+  const onSubmit = (data: {
```
Comment (pKorsholm) [inline src/domain/gift-cards/new.tsx:75]: There should be some data validation somewhere, this throws errors if it's submitted without any denominations

File: src/domain/gift-cards/new.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            </div>
+          </Modal.Header>
+          <Modal.Content>
+            <div className="mb-base">
+              <h3 className="inter-base-semibold">Product information</h3>
+            </div>
+            <div className="flex flex-col gap-y-base">
+              <InputField
+                label={"Name"}
+                required
```
Comment (pKorsholm) [inline src/domain/gift-cards/new.tsx:125]: This `required` is not enforced, the giftcard can be created without inputting anything

**Review by pKorsholm — COMMENTED** [review]
looks very nice! 😍 

I only have comments regarding form validation 😄  (and a bit of generic layout)

File: src/domain/gift-cards/overview.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    })
+
+    window.history.replaceState(baseUrl, "", `?${prepared}`)
+    // TODO: Refresh
+  }
+
+  const actionables = [
+    {
+      label: "Custom Gift Card",
+      onClick: () => console.log("create custom gift card"), // TODO: open modal
```
Comment (zakariaelas) [inline src/domain/gift-cards/overview.tsx:55]: Is this part of another PR? You have a screenshot on your PR description which says "Create Gift Card" but I'm not sure if there is another way to make the modal appear other than this action :sweat_smile: . For now, I just turned the `showCreate` boolean to true to see the create gift card modal :smile:

**Review by zakariaelas — CHANGES_REQUESTED** [review]
Great work Kasper, looks super neat! Left a small comment :+1:

**Review by olivermrbl — APPROVED** [review]
Great work Kap! Think would be great to get merged in latest from `feat/revamp` 👍

File: src/domain/gift-cards/overview.tsx (hunk 4 above)
Comment (olivermrbl) [reply src/domain/gift-cards/overview.tsx:55]: Well spotted @zakariaelas ! This is a separate ticket. Will make sure to create it 👍

Comment (kasperkristensen) [reply src/domain/gift-cards/overview.tsx:36]: The button to show create new Gift Card is found on line 92, and it only shows if you do not have any gift cards in your store. That's properly why you couldn't open it 😄

**Pull request merged**